src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -17,6 +17,21 @@ namespace Microsoft.AspNet.Mvc
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
     public class ProducesAttribute : ResultFilterAttribute, IApiResponseMetadataProvider
     {
+        /// <summary>
+        /// Initializes an instance of <see cref="ProducesAttribute"/>.
+        /// </summary>
+        /// <param name="type">The type of object that is going to be written in the response.</param>
+        public ProducesAttribute([NotNull] Type type)
```

Comment:
It's already settable as a property. The new constructor just lets you create a `[Produces]` with either content types or a Type.